bndtools.m2e/src/bndtools/m2e/MavenImplicitProjectRepository.java

```diff
@@ -69,6 +50,11 @@ public MavenImplicitProjectRepository(IMavenProjectFacade projectFacade, File bn
 		this.bndrunFile = bndrunFile;
 		this.run = run;
 
+		ofNullable(run.getProperties()
+			.get(BndrunContainer.class.getName()))
+			.map(BndrunContainer.class::cast)
+				.ifPresent(asConsumer(bndrunContainer -> fileSetRepository = bndrunContainer.getFileSetRepository()));
```

Comment:
```java
this.fileSetRepository = ofNullable(run.getProperties().get(BndrunContainer.class.getName()))
			.map(BndrunContainer.class::cast)
            .map(BndrunContainer::getFileSetRepository)
            .orElse(null);
```

Seems more straightforward than freaky side effects of settings a field in the ifPresent lambda.

Reply:
done